collector/textfile.go

```diff
@@ -271,10 +301,20 @@ fileLoop:
 		// a failure does not appear fresh.
 		mtimes[f.Name()] = f.ModTime()
 
-		for _, mf := range parsedFamilies {
+		for _, metricFamily := range parsedFamilies {
+			metricFamilies = append(metricFamilies, metricFamily)
+		}
+	}
+
+	if duplicateMetricEntry(metricFamilies) {
+		log.Errorf("Duplicate metrics detected in files")
+		error += 1.0
+	} else {
+		for _, mf := range metricFamilies {
 			convertMetricFamily(mf, ch)
 		}
 	}
+	// If crash is caused by convertMetricFamily above, move to an else {} block here
```

Comment:
What does this mean?

Reply:
Oops, this was a leftover TODO when I was writing the change. I've removed that now.